### PR TITLE
Share buttons and Twitter Cards

### LINKED ISSUE #677: Update sharing to latest Twitter Card
**User Story:**

As a user, I want to share Mapstory content on Twitter visually, so that the viewer of my tweet sees animation that my mapstory content intends to show.

**Acceptance Criteria**

- From a mapstory detal page, add a sharing option for Tweet This.
- Configure mapstory media output (i.e. Gif? Video?) to align with Twitter card policy, so that visual appears similar to examples linked below.

**Workflow**

- When user clicks "Share this" button on mapstory detail page they currently have options to share link or copy/past embed. Under these two options we can add "Tweet". Clicking tweet should bring up the "Share a link on Twitter" twitter dialogue with a pre-loaded but editable tweet. Pre-loaded tweet should have: MapStory Title, link, via @MapStory by StorytellerName and animated thumbnail in appropriate format.

See below for a depiction of this workflow in Youtube, which is similar

screen shot 2017-07-07 at 11 28 53 am

**Notes**

Work was done on Twitter Cards on retro. Here's a link to the issue and discussion for reference:

https://github.com/MapStory/mapstory-meta/issues/500

Twitter Card docs for developers can be found at https://dev.twitter.com/cards/overview.

Examples of animated maps shared on Twitter:

https://www.youtube.com/watch?v=o5dQKDtfb28&feature=youtu.be
http://lenkiefer.com/2016/08/24/more-maps
https://twitter.com/CSIS/status/828670600461692928/photo/1

Here is a single chapter mockup produced by @reynalddizon:

www gifcreator me_1knskj

Here's a multiple-chapter mockup from Rey Dizon at super-small size
stories_gif_with_annot_sm

And a multiple-chapter mockup at i-phone size:
stories_gif_with_annot_lg

Here is a breakdown of the elements the Twitter Card media output would include (right side column):

screen shot 2017-05-05 at 5 11 31 pm

### PR DESCRIPTION
**Pull request by josellausas**
Closes #677 

This PR:

- Adds meta tags needed for using Twitter cards when sharing a Layer and a MapStory.
- Adds social buttons for Layer sharing.
- Adds social buttons for Story sharing.

New Features
--------------

- Sharing a MapStory detail URL on Twitter will display a 'Summary Card' 
- Sharing a Layer detail URL on Twitter will display a 'Large Image Card'
- Unit tests for Twitter meta tags.

### CODE REVIEW

**Comment by josellausas** [comment]
Added social icons for layers.
Story social icons WIP

screen shot 2017-07-12 at 3 16 07 pm

**Comment by josellausas** [comment]
Added social share buttons for Stories.

screen shot 2017-07-12 at 4 41 49 pm

screen shot 2017-07-12 at 4 41 58 pm

File: mapstory/static/style/site/share-icons.less [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+.share-btn:active {
+  position: relative;
+  top: 2px;
+  box-shadow: none;
+  color: #e2e2e2;
+  outline: none;
+}
+
+.share-btn.twitter     { background: #55acee; }
```
Comment (emilyashley) [inline mapstory/static/style/site/share-icons.less:25]: For site-wide consistency and maintainability, we've got some less variables [here](https://github.com/MapStory/mapstory/blob/master/mapstory/static/style/site/variables.less#L16) with the standard social media colors, feel free to add those that aren't listed into this variables.less file as well

**Comment by Coop56** [comment]
I've pulled this branch down and I'm getting ready to test and review it today.

**Review by Coop56 — APPROVED** [review]

**Comment by emilyashley** [comment]
Note to self: PRESS THE SUBMIT REVIEW BUTTON 😝

**Pull request merged**